agent/internal/command.go

```diff
@@ -69,25 +77,31 @@ func (m *CommandManager) Do(commands []api.CommandRequest) map[string]api.Comman
 		results[cmd.Uuid] = result
 	}
 
-	if len(results) == 0 {
-		return nil
-	}
 	return results
 }
 
-func (m *CommandManager) InstrumentationEnable() api.CommandResult {
-	err := m.agent.InstrumentationEnable()
-	return commandResult(err)
+func (m *CommandManager) InstrumentationEnable([]json.RawMessage) error {
+	return m.agent.InstrumentationEnable()
 }
 
-func (m *CommandManager) InstrumentationRemove() api.CommandResult {
-	err := m.agent.InstrumentationDisable()
-	return commandResult(err)
+func (m *CommandManager) InstrumentationRemove([]json.RawMessage) error {
+	return m.agent.InstrumentationDisable()
 }
 
-func (m *CommandManager) ActionsReload() api.CommandResult {
-	err := m.agent.ActionsReload()
-	return commandResult(err)
+func (m *CommandManager) ActionsReload([]json.RawMessage) error {
+	return m.agent.ActionsReload()
+}
+
+func (m *CommandManager) IPSWhitelist(args []json.RawMessage) error {
+	if argc := len(args); argc != 1 {
+		return fmt.Errorf("unexpected number of arguments: expected 1 argument but got %d", argc)
```

Comment:
Will be converted to agent_exception right?

Reply:
No, as commands result messages. I need to add a call to the Error logger in order to put it into the event batch as an exception.

Will be fixed after the sequence of merges I start.

Thanks for pointing this out.